cpu/esp_common/include/irq_arch_common.h

```diff
@@ -67,6 +67,79 @@ extern volatile uint32_t irq_interrupt_nesting;
 #define critical_exit_var(m)    irq_restore(m)
 /** @} */
 
+/**
+ * @brief Disable all maskable interrupts
+ */
+__attribute__((always_inline)) static inline unsigned int irq_disable(void)
+{
+    uint32_t state;
+
+    /* read and set interrupt level (RSIL) */
+    __asm__ volatile (
+        /* this will block IRQs with prio <= intlevel */
+        "rsil %[state], %[intlevel]"
+        : [state]       "=a" (state)
+        : [intlevel]    "I"(15) /* 15 is highest prio, blocking all IRQs */
```

Comment:
```suggestion
        : [intlevel]    "I"(XCHAL_EXCM_LEVEL) /* 15 is highest prio, blocking all IRQs */
```

Reply:
We should use a constant from the SOC definition here. We could use `XCHAL_NUM_INTLEVELS` to disable all interrupts or `XCHAL_EXCM_LEVEL` to disable only low level interrupts.

ESP32:
```
#define XCHAL_EXCM_LEVEL		3
#define XCHAL_NUM_INTLEVELS		6
```
ESP8266:
```
#define XCHAL_EXCM_LEVEL		1
#define XCHAL_NUM_INTLEVELS		2
```